### PR TITLE
Enable subsolver tolerances tuning, fix negative tolerance

### PR DESCRIPTION
**Pull request by geoffroyleconte**
I changed the algorithms so that it is possible to set some specific tolerances in the `subsolver_options`. I did a slight modification in the default `subsolver_options` so that it keeps the existing behaviour.

It allowed me to find better parameter for the benchmarks. 

I also updated the stopping criterion in TRDH so that it handles negative (but small) `ξ1` or `ξ` values, and fixed what I think is a bug in R2 with the `neg_tol` parameter (which is always positive).

### CODE REVIEW

File: benchmarks/tables/bpdn-constr-table.jl [hunk 1]
```diff
@@ -6,18 +6,20 @@ compound = 1
 model, nls_model, sol = bpdn_model(compound, bounds = true)
 
 f = LSR1Model(model)
-λ = norm(grad(model, zeros(model.meta.nvar)), Inf) / 10
+λ = 1.0e-1 # norm(grad(model, zeros(model.meta.nvar)), Inf) / 10
```
Comment (dpo) [inline benchmarks/tables/bpdn-constr-table.jl:9]: Are the results very different if you  leave the default lambda ?

File: benchmarks/tables/bpdn-constr-table.jl [hunk 2]
```diff
@@ -6,18 +6,20 @@ compound = 1
...
 h = NormL1(λ)
 
 verbose = 0 # 10
-ϵ = 1.0e-6
+ϵ = 1.0e-5
```
Comment (dpo) [inline benchmarks/tables/bpdn-constr-table.jl:13]: Is this one used?

**Review by dpo — COMMENTED** [review]
Thank you! Can we use TRDH inside LM and LMTR now?

File: benchmarks/tables/bpdn-constr-table.jl (hunk 1 above)
Comment (geoffroyleconte) [reply benchmarks/tables/bpdn-constr-table.jl]: Not that much, I updated the tables and the benchmarks in the article.

**Comment by geoffroyleconte** [comment]
> Thank you! Can we use TRDH inside LM and LMTR now?

Not at the moment be this would be a very minor change to make them work. I can do it in my next PR once this one is merged.

File: benchmarks/tables/bpdn-constr-table.jl [hunk 3]
```diff
@@ -7,51 +7,101 @@ model, nls_model, sol = bpdn_model(compound, bounds = true)
...
-  ν = 1.0,
+  ROSolverOptions(ν = ν, ϵa = ϵ, ϵr = ϵ, verbose = verbose, maxIter = maxIter, spectral = true)
+optionsbis =
+  ROSolverOptions(ν = ν, ϵa = ϵ, ϵr = ϵ, verbose = verbose, maxIter = maxIter, spectral = true, reduce_TR = false)
+options2 = ROSolverOptions(spectral = false, psb = true, ϵa = ϵi, ϵr = ϵri, maxIter = maxIter_inner)
+options2bis = ROSolverOptions(spectral = false, psb = true, ϵa = ϵi, ϵr = ϵri, maxIter = maxIter_inner, reduce_TR = false)
+options3 = ROSolverOptions(spectral = false, psb = false, ϵa = ϵi, ϵr = ϵri, maxIter = maxIter_inner)
+options3bis = ROSolverOptions(spectral = false, psb = false, ϵa = ϵi, ϵr = ϵri, maxIter = maxIter_inner, reduce_TR = false)
+options4 = ROSolverOptions(spectral = true, ϵa = ϵi, ϵr = ϵri, maxIter = maxIter_inner)
+options4bis =
```
Comment (dpo) [inline benchmarks/tables/bpdn-constr-table.jl:28]: Let's give all these options descriptive names and avoid "bis" (it's French and will confuse non-French speakers).

**Review by dpo — APPROVED** [review]

**Pull request merged**